### ISSUE TITLE
Provide filter by kuma.io/service on */external-services endpoints

### ISSUE THREAD

**Issue by kleinfreund** [kumahq/kuma#5908]
TL;DR: We need a way to retrieve an ExternalService given a **ServiceInsight name** (which can be different than the ExternalService name) in order to show it in the service detail view.

### Description

In order to be able to look up external services by service name (which might be different than the _external_ service name), a filter on the */external-services endpoints will be needed.

This mechanism is needed for the service list and detail views in the GUI where we list services (including external services) based on the service insights API. In order to correctly display data for external services, we need to be able to reference an external service from data that exists on the service insights API.

```sh
curl 'http://localhost:5681/meshes/default/external-services?tag=kuma.io/service:httpbin'
```

This is meant to provide a proper fix for https://github.com/kumahq/kuma-gui/issues/599.

**Comment by jakubdyszkiewicz** [comment]
> Triage: provide same filters as for regular DPPs

**Comment by lahabana** [comment]
I'm extending this with a way to add filters to any resource type.
